frontend/src/components/common/ImpactPerCause.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect } from "react";
+import { Box, Text, Flex, Button, Center, Circle } from "@chakra-ui/react";
+
+
+interface ItemData {
+  item_id: number;
+  item_name: string;
+  total_impact: number;
+}
+
+interface CauseData {
+  cause_id: number;
+  cause_name: string;
+  items: ItemData[]; 
+}
+
+const ImpactPerCause: React.FC = () => {
+  const [pageIndex, setPageIndex] = useState(0);
+  const [causes, setCauses] = useState<CauseData[]>([]);
+
+  // Buttons to move between different causes/pages
+  const handlePrevious = () => {
+    setPageIndex((prev) => (prev > 0 ? prev - 1 : prev));
+  };
+
+  const handleNext = () => {
+    setPageIndex((prev) => (prev < causes.length - 1 ? prev + 1 : prev));
+  };
+
+  // As of right now, impact is fetched from this link. In the future, this will be replaced by authContext.
+  // Change to localhost:5000 and modify the userId below based on the data entered in Prisma
+  async function fetchImpactData(userId: string) {
+    try {
+      const response = await fetch(`http://localhost:5001/impacts/${userId}`);
+      const data = await response.json();
+      console.log(data);
+      return data;
+    } catch (error) {
+      console.error("Error fetching impact data,", error);
+      return null;
+    }
+  }
+
+  useEffect(() => {
+    // Hardcoded in for testing purposes, change depending on what your user
+    const userId = "cly144mky0000bntg3dupxlx1";
+    fetchImpactData(userId).then((data) => {
+      setCauses(data);
+    });
+  }, []);
+
+  return (
+    <Box
+      maxW="40rem"
+      mx="auto"
+      p={5}
+      borderWidth="1px"
+      borderRadius="lg"
+      overflow="hidden"
+      borderColor="#E0DCDA"
+    >
+      {/* Header text displaying the cause name or loading message */}
+      <Text
+        fontSize="2xl"
+        mt={3}
+        mb={0}
+        textAlign="center"
+        fontWeight="bold"
+        color="#4D4D4D"
+      >
+        {causes.length > 0
+          ? `Total Impact For ${causes[pageIndex].cause_name}`
```

Comment:
Can you guys store strings (e.g. "Total Impact For") in a JSON file and load them dynamically (instead of hardcoding them in the frontend components)?

We don't currently have a JSON file defined for frontend strings, so we'll have to create one.

At the end of the day, the frontend of this app is going to need to be in French. So, having all our strings in a JSON file will allow for easy translation.

Let me know if you guys have any questions, thanks for doing this 🙏